Sources/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -306,46 +311,47 @@ extension ProfileStore {
 	}
 
 	func assertDeviceOwnsSnapshotElseCreateNew(_ snapshot: ProfileSnapshot) async throws {
-		do {
-			try await Self.checkIfDeviceOwnsProfileSnapshot(snapshot.header)
-		} catch {
-			// Note: We do not reset the active profile id, as doing so, will imply that user has no profile.
-			//       Instead, we will prompt that the user, that the currently active profile is used on other device.
-
-			// Go to ephemeral state straightaway. The Wallet will redirect user to the Onboarding screen.
-			await changeState(to: .ephemeral(.init(
-				profile: Self.newEphemeralProfile(),
-				loadFailure: .profileUsedOnAnotherDevice(error as! Profile.ProfileIsUsedOnAnotherDeviceError) // safe cast
-			)))
-
-			// rethrow the error to halt the execution up the chain
-			throw error
-		}
-	}
-
-	static func checkIfDeviceOwnsProfileSnapshot(_ header: ProfileSnapshot.Header) async throws {
+		await Self.checkIfDeviceOwnsProfileSnapshot(snapshot)
+		// FIXME: Reintroduce later
+		//        do {
+//		} catch {
+//			// Note: We do not reset the active profile id, as doing so, will imply that user has no profile.
+//			//       Instead, we will prompt that the user, that the currently active profile is used on other device.
+//
+//			// Go to ephemeral state straightaway. The Wallet will redirect user to the Onboarding screen.
+//			await changeState(to: .ephemeral(.init(
+//				profile: Self.newEphemeralProfile(),
+//				loadFailure: .profileUsedOnAnotherDevice(error)
+//
+//			// rethrow the error to halt the execution up the chain
+//			throw error
+//		}
+	}
+
+	// The implementation of this is not what we want in the future, it is
+	// written like this to rectify bad state for some users who might incorrectly
+	// have a discprenacy between their `profile.header.lastUsedOnDevice.id` and
+	// the `deviceIdentifer` saved in keychain
+	static func checkIfDeviceOwnsProfileSnapshot(_ profileSnapshot: ProfileSnapshot) async {
```

Comment:
this function could be named `tryRectifyingPotentialDeviceIDDisrepancy`.... which is what it does now.